The ultimate goal of computer assisted drug design is to improve rational drug design by exploiting the continuously increasing processing power available both in high performance super computers as well as in single workstations.
Researchers seek to supplement their ability to quickly examine a large number of possible interactions or gain insights that would be much more difficult, both in terms of time and expense, to obtain through biochemical experiments thorugh application of computational methods to drug design.
Different classes of programs have been developed to help solve each of the three distinct steps in the pre-clinical stages of drug development:
\begin{enumerate}
\item Hit Identification -- the process of screening a large small molecule database, containing up to one million and sometimes more small molecules, to identify small molecules that bind a given target protein, or hits.
These hits are usually small molecules with a target binding affinity on the order of micromolars.
\item Hit to lead optimization -- the process of modifying these hit molecules, either by substitution or addition of chemical moieties or mixing and matching substructures between given hits, to produce compounds with higher binding affinities than the initial hit compounds.
Hit to lead optimization seeks to improve the micromolar binding affinity of hit compounds to nanomolar affinity or better.
\item Lead Optimization -- the final step of modifying lead compounds to increase ``druglikeness'' to ensure that the molecule is sufficiently soluble, well tolerated, and does not disrupt regular cellular function.
\end{enumerate}

\subsubsection{Hit Identification}
\label{subsubsection:hit_identification}
% hit identification in general
The earliest form of hit identification experiments were animal screens, where mutant animals were studied to find the specific gene or protein involved in a specific phenotype.
This type of experiment relies on careful genetic controls and breeding, but also some element of luck in observing a relevant phenotype in the first place.
``Brute force'' animal screens have since been improved with extensive mutation libraries and exhaustive non-lethal mutation libraries for organisms such as yeast and {\it Escherichia coli}.
Even so, these screens are slow, often taking three years or longer, and any such studies in mammalian model organisms, like mice, quickly become extremely expensive.
Furthermore, these screens can be error prone, as performing a large number of repetitive experiments causes even the most fastidious of scientists to lose focus.
High-throughput screening seeks to supplement the human factor with robots, which are capable of performing similar experiments with greater speed and fewer errors.
With the help of this automation it is possible to test the interactions of as many as 100 million different reactions per day \cite{agresti2010ultrahigh}.
However, the high initial cost of high-throughput screening equipment as well as the cost of the small molecule libraries necessary for screening are often prohibitive even to large research institutions.
In order to make this sort of experiment available to a larger number of institutions, some research institutions have instituted means of sharing this equipment, through high-throughput screening as a service type arrangements \cite{htsrc,mssr}.

% computational equivalents of hit identification
The direct computational equivalent to high-throughput screening is virtual screening, where a library of small molecules is computational ``docked'' into the active site of the target protein, and some scoring metric is used to identify possible binders.
In this sort of computational screen, the problem of the cost of small molecule libraries is essentially solved, as there are readily available libraries, some of which are free, of drug-like small molecules for use in virtual screening programs.
For example, ZINC is a free database that provides a library of over seven-hundred thousand commercially available small molecules in a number of different file formats for use in virtual screening \cite{irwin2005zinc}.

% fragment assembly methods
Another possibility for hit identification {\it in silico} is through fragment assembly methods.
These methods seek to identify conserved moities in various binding compounds and assemble a high affinity binder by joining together these moities in a single compound which collects the binding affinities of each of its constituent parts \cite{jorgensen2006computer,jorgensen2004many,jorgensen2009efficient}.

% early docking history
The first published study using computational docking dates to 1982, by Irwin Kuntz, describing a program that would later go on to become the well known DOCK program \cite{kuntz1982geometric}.
Generally, docking consists of a method of quickly screening possible protein-small-molecule interaction conformations.
An emphasis is placed on the computational cost of evaluating the energy function over accuracy, as the poses generated by this step are usually fed into structural refinement programs for further sampling and more accurate estimation of energies.
For example, in the original Kuntz study, the system only only had six degrees of freedom on which to sample -- three translational and three rotational degrees of freedom for the ligand with the protein held fixed.
Along with a hard sphere collision model, this provided a sufficiently selective screen to identify the native binding geometry of the heme group to myoglobin as well as thyroid hormone analogs to prealbumin \cite{kuntz1982geometric}.

% growth of pdb data
The Protein Data Bank (PDB) is a commonly used source of structural information used in screens for hit compounds \cite{abola1984protein}. 
The rate at which new structures are being deposited into the PDB is increasing on an annual basis, see figure \ref{figure:pdb_growth}.
But new tools are necessary to draw meaningful insights, hopefully leading to new drugs, from this wealth of data.
\begin{figure}[H]
\begin{center}
\includegraphics[width=\textwidth]{figures/pdb_deposit_rate.png}
\caption{The rate at which new structures are deposited into the PDB over the last two decades.
Due to a variety of improvements in the field of crystallography, this rate has been steadily increasing.
Plot generated using data from the PDB \protect\cite{berman2003announcing,berman2000protein}.}
\label{figure:pdb_growth}
\end{center}
\end{figure}

% expected future growth of pdb
For example, a recent advancement in the field of crystallography is ``crystal-less'' crystallography, in which small molecules are bound by a porous scaffold matrix.
The regular structure of the matrix imparts a regular packing arrangement, necessary for interpreting diffraction patterns, onto the arrangement of small molecules.
This has the potential to address one of the largest difficulties in obtaining quality structural data for proteins, which is that it is very difficult to purify and crystallize certain proteins \cite{inokuma2013x}.

% unsorted
The number of target molecules of the set of all drugs currently available on the market consists of only about five-hundred proteins.
The bottleneck in the introduction of new chemical entities is not virtual screening, but rather optimizing these hits into higher affinity leads (see \ref{subsubsection:hit_to_lead}) and eventually balancing the requirements across all characteristics to produce a new drug (see \ref{subsubsection:lead_optimization}) \cite{bleicher2003hit}.

% drug targets
Most disease implicated proteins are not targeted by current drugs and finding improved drugs for those proteins which are currently drug targets can be very difficult, and sometimes not productive. 
Therefore, new chemical entities frequently aim to target proteins that are currently not targetd by currently available drugs. 
Of the entire proteome, only \textapprox30,000 proteins are regulated by small molecule binding, making them reasonable targets of drug action.
A large number of these possible drug targets are not implicated in any disease.
Due to this and a number of other factors, estimates of the total number of the proteins regulated by small molecule binding that are possible drug targets is much lower than 30,000.
Frequently cited numbers for the number of possible drug targets in humans are six-hundred to fifteen-hundred, still significantly higher than the total number of targets exploited by current drugs \cite{imming2006drugs,overington2006many}.
Further, different families of cellular proteins are not equally likely to be targets of drugs.
As of 2002, 47\% of current drug targets are enzymes, followed by 30\% being GPCR's \cite{hopkins2002druggable}.

% drug-likeness
After identifying appropriate proteins as drug targets, focus is then turned to assessing the drug-likeness of candidate small molecules.
A number of key characteristics are generally true of drug-like small molecules.
These characteristics are often referred to as the ``rule of five'' \cite{lipinski2004lead,hopkins2002druggable}:
\begin{enumerate}
\item Five or fewer hydrogen bond donors,
\item 500 Da or less total molecular mass,
\item high liphophilicity,
\item number of nitrogen and oxygen atoms is not greater than 10 \cite{rule_of_five}.
\end{enumerate}
Therefore, when screening small molecules libraries for hits, these criteria can help focus efforts on small molecules which are likely to have good absorption, distribution, metabolism, and excretion, characteristics, which are essential in effective drugs.
These criteria are discussed further in \nameref{subsubsection:lead_optimization} (\ref{subsubsection:lead_optimization}).

% flexibility
In addition to the drug-likeness of small molecules, another desirable criteria is some degree of flexibility of the lead compound.
This is especially important as the location of heavy atoms in the target protein is frequently only known to an accuracy of \textapprox0.4 angstroms and flexible substrates can alter their conformations in order to create better contacts with the protein structure, thereby increasing binding affinity.
% early example
One of the earliest examples of the successful application of structure based drug design is the carbonic anhydrase inhibitor dorzolamide, in which many of these ideas were applied to find a drug with very high binding affinity \cite{greer1994application}.
Through understanding the protein-ligand conformation and specific contacts, Greer {\it et. al.} were able to modify a known substrate into an effective drug. 

%Further, specific knowledge of the binding geometry between the initial lead compound and the target makes it possible to computationally screen possible chemical group substituents, to maximize binding affinity.s

% limitations
Despite advantages in speed and cost, computational screening has struggled to produce the same results as empirical screening, due to limitations in accuracy.
However, more recently virtual screening has succeeded in producing hit rates greater than those from empirical screening techniques.
Virtual screening has been used to identified leads which were later developed into the human immunodeficiency virus (HIV) protease inhibitor Viracept (figure \ref{figure:nelfinavir}) \cite{kaldor1997viracept}, and the anti-influenza drug Relenza \cite{von1993rational,shoichet2004virtual}.

\begin{figure}[h]
    \centering
    \begin{subfigure}[b]{0.3\textwidth}
        \centering
        \includegraphics[width=\textwidth]{figures/nelfinavir_small.png}
        \label{fig:nelfinavir_chemical}
        \caption{}
    \end{subfigure}%
    \hspace{0.1\textwidth}
    \begin{subfigure}[b]{0.3\textwidth}
        \centering
        \includegraphics[width=\textwidth]{figures/complexed_nelfinavir_white.png}
        \label{fig:nelfinavir_docked}
        \caption{}
    \end{subfigure}
    \caption{(a,b) The HIV protease inhibitor, nelfinavir, marketed under the name Viracept was originally identified using a computational docking screen.
It has a very high binding affinity (2 nM) for HIV-1 protease, its target protein.
(b) Here it is shown crystallized with multidrug variant (ACT) (V82T/I84V) of HIV-1 protease, PDBid 3EL5 \protect\cite{king2012extreme}.
Generated with Visual Molecular Dynamics \protect\cite{humphrey1996vmd} and \protect\cite{povray}.}
\label{figure:nelfinavir}
\end{figure}

A number of challenges limiting the utility of docking programs have been identified:
\begin{enumerate}
    \item The number of possible small molecules is essentially unbounded.
    However, only a very small fraction of these ligands are potential drug compounds.
    Limiting sampling to this subspace of small molecules is a challenging problem, but would significantly enhance accuracy and efficiency of virtual screening.
    \item The number of conformations of ligand molecules rises exponentially with the number of internal degrees of freedom of the ligand.
    Sampling the huge conformational space of the ligand becomes extremely computationally expensive.
    \item Accurately assessing or comparing the energy of different protein-ligand complexes or conformations \cite{shoichet2004virtual}.
    This is because molecular mechanics force fields must make approximations about the molecular forces.
\end{enumerate}

% similarity between hits and leads
It has been found that introduced drugs are often very chemically similar the hit compounds from which they were derived \cite{proudfoot2002drugs}.
Therefore, in order to increase the diversity of drugs and find drugs that are able to treat new diseases, or diseases that have evolved resistance to current drugs, it may be necessary to either increase the size of screened databases or increase the possible diversity which might be introduced through the hit-to-lead step.

\subsubsection{Hit-to-Lead Optimization}
\label{subsubsection:hit_to_lead}
\input{1_intro/drug_development/hit_to_lead}

\subsubsection{Lead Optimization}
\label{subsubsection:lead_optimization}
In lead optimization the compounds identified by the earlier steps in the process are optimized to drug molecules.
Though the objectives of lead optimization overlap heavily with those of the hit-to-lead stage, the compounds at the lead optimization stage are much more plausible as potential drugs.
While this can include further increasing binding affinity to the target protein, usually the focus of lead optimization is on other characteristics including selectivity, ease of synthesis, pharmacokinetic properties and intellectual property concerns \cite{keserHu2006hit}.
Computational modeling can not only help identify hit compounds and convert those initial hits into leads, but also help estimate and optimize absorption, distribution, metabolism, elimination, toxicology, sometimes referred to as the ADME characteristics \cite{kerns2008drug}.
Computational models for ADME characteristics usually use regression equations or neural networks to predict these characteristics \cite{jorgensen2004many}.

Up to one half of all drugs which do not survive clinical trials fail to do so because of lack of efficacy, which is influenced both by binding affinity and the ADME characteristics of the molecule.
The number of drugs that fail to make it through clinical trials due to toxicity is similarly high, about 40\% \cite{li2001screening}.
Advancing a potential drug to clinical trials represents a very large financial investment, and effective computational screens of lead molecules at this point in the process can reduce the rate of failure in clinical trials, thereby having a very large impact on the final costs of new drugs brought to market.
